test/integration/studios-page.test.js

```diff
@@ -52,5 +68,83 @@ describe('studio page while signed out', () => {
         let descriptionText = await studioDescription.getText();
         await expect(descriptionText).toEqual('a description');
     });
+});
+
+describe('studio management', () => {
+    // These tests all start on the curators tab of a studio and signed out
+
+    beforeAll(async () => {
+        driver = await buildDriver('www-integration studio management');
+        await driver.get(rootUrl);
+
+        // create a studio for tests
+        await signIn(username2, password, driver);
+        await findByXpath('//span[contains(@class, "profile-name")]');
+        await driver.get(rateLimitCheck);
+        await driver.get(myStuffURL);
+        await clickXpath('//form[@id="new_studio"]/button[@type="submit"]');
+        await findByXpath('//div[@class="studio-tabs"]');
+        promoteStudioURL = await driver.getCurrentUrl();
+        curatorTab = await promoteStudioURL + 'curators';
+    });
 
+    beforeEach(async () => {
+        await clickXpath('//a[contains(@class, "user-info")]');
+        await clickText('Sign out');
+        await driver.get(curatorTab);
+        await findByXpath('//div[@class="studio-tabs"]');
+    });
+
+    afterAll(async () => await driver.quit());
+
+    test('invite a curator', async () => {
+        // sign in as user2
+        await signIn(username2, password, driver);
+        await findByXpath('//span[contains(@class, "profile-name")]');
+
+        // invite user3 to curate
+        let inviteBox = await findByXpath('//div[@class="studio-adder-row"]/input');
+        await inviteBox.sendKeys(username3);
+        await clickXpath('//div[@class="studio-adder-row"]/button');
+        let inviteSuccess = await findByXpath('//div[@class="alert-msg"]'); // the confirm alert
+        let inviteSuccessVisible = await inviteSuccess.isDisplayed();
+        await expect(inviteSuccessVisible).toBe(true);
+    });
+
+    test('accept curator invite', async () => {
+        // Sign in user3
+        await signIn(username3, password, driver);
+        await findByXpath('//span[contains(@class, "profile-name")]');
+
+        // accept the curator invite
+        await clickXpath('//button[@class="studio-invitation-button button"]');
+        let acceptSuccess = await findByXpath('//div[contains(@class,"studio-info-box-success")]');
+        let acceptSuccessVisible = await acceptSuccess.isDisplayed();
+        await expect(acceptSuccessVisible).toBe(true);
+    });
+
+    test('promote to manager', async () => {
+        // sign in as user2
+        await signIn(username2, password, driver);
+        await findByXpath('//span[contains(@class, "profile-name")]');
+        // for some reason the user isn't showing up without reloading the page
+        await driver.get(curatorTab);
+
+        // promote user3
+        let user3href = '/users/' + username3;
+        // click kebab menu on the user tile
+        let kebabMenuXpath = '//a[@href = "' + user3href + '"]/' +
```

Comment:
in the future you might consider using template strings if that makes it simpler

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals